homeassistant/components/met/__init__.py

```diff
@@ -173,7 +175,9 @@ def set_coordinates(self) -> bool:
 
     async def fetch_data(self) -> MetWeatherData:
         """Fetch data from API - (current weather and forecast)."""
-        await self._weather_data.fetching_data()
+        resp = await self._weather_data.fetching_data()
+        if not resp:
+            raise asyncio.TimeoutError
```

Comment:
Hmm, this is an `asyncio` error, which is kinda awkward to raise here? As in, that is not managed by us.

Maybe raise a custom error (from HomeAssistantError) or `UpdateFailed` instead?

Reply:
As far as I understand this update function is used only here: https://github.com/home-assistant/core/blob/97a04dc6225e485c3e9e6f82b3e8b79f3979bef7/homeassistant/components/met/__init__.py#L109

I think it would be better to move this fetching part directly there and raise `raise UpdateFailed` instead of raising here and raising again. 
Another option is to change the return type to allow `None`